### ISSUE TITLE
name resolution failure

### ISSUE THREAD

**Issue by mbryde** [bubuntux/nordvpn#297]
**Describe the bug**
After a period of time the container for some reason stops resolving dns queries. I have tried both with and without providing it dns servers. It usually happens after around 4 hour and seems to be pretty consistent.

**To Reproduce using docker CLI**
Full command needs to be provided (hide credentials)
`docker run --name NordVPN -d -p 8118:8118 --cap-add=NET_ADMIN -e USER=<username> -e PASS=<password> -v /etc/localtime:/etc/localtime -e TECHNOLOGY=NordLynx -e NETWORK=192.168.1.0/24 bubuntux/nordvpn `

**Expected behavior**
The docker container should always resolve dns queries no matter what.

**Logs**
2021-12-12 21:16:16.624 149a865f2700 Info: Can not resolve www.google.com: Temporary failure in name resolution
2021-12-12 21:16:16.624 149ab50aa700 Info: Can not resolve www.google.com: Temporary failure in name resolution
2021-12-12 21:16:16.871 149a871f8700 Info: Can not resolve www.google-analytics.com: Temporary failure in name resolution

**Comment by Rickexe** [comment]
How did you manage to get that far? xD
I can't even start the container because logs say invalid username and password login deprecated. It seems on other forums that pretty much everyone has this issue.

**Comment by mbryde** [comment]
> How did you manage to get that far? xD I can't even start the container because logs say invalid username and password login deprecated. It seems on other forums that pretty much everyone has this issue.

Deprecated login with username and password? Sounds weird. Never had that problem.

**Comment by Rickexe** [comment]
Loolll... can you share your container setup?

**Comment by Rickexe** [comment]
![image](https://user-images.githubusercontent.com/48714886/145893118-eecb4444-69fa-4a8c-8243-dd95dd4ca97c.png)

**Comment by mbryde** [comment]
Are you sure you are running the latest version of the docker image?

**Comment by Rickexe** [comment]
I am using the latest tag yes. With this run command (with my credentials obviously). I get that error. How did you fill in your credentials? just email and password or the ones from nordvpn.com when you log in? 

docker run -ti --cap-add=NET_ADMIN --name vpn \
           -e USER=user@email.com -e PASS='pas$word' \
           -e TECHNOLOGY=NordLynx -d ghcr.io/bubuntux/nordvpn

**Comment by derchan** [comment]
I get the same error since yesterday. Did something change on the nord vpn side.

**Comment by someguy3013** [comment]
Seeing the same error on my end. Nothing has changed with my config, username, or password. 

2021/12/15 16:32:22 Get "https://zwyr157wwiu6eior.com/v1/helpers/ips/insights": quic: performing request: curl: Couldn't connect to server
[services.d] done.
Password login is deprecated.
'nordvpn login --nordaccount' will become the default login method in the future.
|2021/12/15 16:32:27 [Error] logging in: default api: some of X-Digest, X-Authorization headers are missing
It's not you, it's us. We're having trouble reaching our servers. If the issue persists, please contact our customer support.
Invalid Username or password.
[cmd] /bin/sh exited 1
[cont-finish.d] executing container finish scripts...
[cont-finish.d] done.
[s6-finish] waiting for services.
[s6-finish] sending all processes the TERM signal.
[s6-finish] sending all processes the KILL signal and exiting.

**Comment by dsgnr** [comment]
I've tested this on numerous devices/package versions (outside of Docker). Something is up with their upstream API afaics. I spoke with their support and they say its something they're aware of, but unable to provide an incident/status page to follow.

Inside Docker (standard Ubuntu 20), I get the above errors. On my test VM (same OS), I get a password failure:
~~~
\2021/12/15 16:44:19 [Error] logging in: Unauthorized: Invalid password
It's not you, it's us. We're having trouble reaching our servers. If the issue persists, please contact our customer support.
~~~

Advise given was to configure OpenVPN...

**Comment by TheWhiteWarlock** [comment]
Been ripping my hair out, very new to docker and setup multiple containers on a Synology, had to reboot my NAS tonight and my NordVPN container died with "Invalid username and password" after working for a week without issue. Hours later I have reset my account at Nord and still no joy. Started using "azinchen/nordvpn" and that works a treat without any connection issues.

No Idea what the connection is with this image and Nord but I am glad that it is not just me and other people are having issues.

**Comment by dsgnr** [comment]
I don't believe this is a Docker/image issue since I've tried on numerous VM's on various Linux distros using the nordvpn package from their repo and I've been receiving the same issue.

However, saying that - everything is now working fine for me (inside and outside Docker) 🥳

**Comment by TheWhiteWarlock** [comment]
I will test again tomorrow to confirm but I agree that the issue looks upstream, after spending hours on this just want a glass of wine now.

**Comment by bubuntux** [comment]
i would recommend migrate to https://github.com/bubuntux/nordlynx when possible

**Comment by TheWhiteWarlock** [comment]
Thank you, I assume it as easy to setup as azinchen/nordvpn and and bubuntux/nordvpn which when they work were not to much trouble.

**Comment by TheWhiteWarlock** [comment]
Checked again today and still getting invalid username or password error message in the log files.

**Comment by akagami-x** [comment]
having the same issue i asked nordvpn support and recived this answer

Unfortunately, since 3.12.2 update you need to log in through the browser.

so headless servers and docker would need a new way to lwork around this

**Comment by TheWhiteWarlock** [comment]
Well, at least we have an answer, just a shame it is not the one I wanted to hear, I have 2 years left on PIA might jump back over to them again. Only been with Nord for 3 weeks :(

**Comment by akagami-x** [comment]
i spun up a dektop ubuntu vm, loged into the nordvpn app with the web auth retrievied the nordlynx privatekey with wg show nordlynx private-key and switched to bubuntux/nordlynx  everything is back up and running like a charm

**Comment by Rickexe** [comment]
Ok, I'll try the nordlynx container tomorrow. Suddenly tons of people responding to this issue xD. Nice :)

**Comment by 3dplumber** [comment]
Nord confirmed an issue with MFA. If you dissable MFA on nord account the container will log in as before.

**Comment by TheWhiteWarlock** [comment]
I just tested and confirmed that it is working again with MFA disabled as reported above

**Comment by LaStrada** [comment]
I've tested this a few times now, but I still have this issue even after disabling MFA 😞 Will test again a bit later

**Comment by Rickexe** [comment]
I have never had MFA enabled and it doesn't work for me.

**Comment by 3dplumber** [comment]
We may be mixing two different issues.
Worth a try to change your password and leave MFA off for test.

You can change password back again after.

**Comment by akagami-x** [comment]
alternative method to login:
Run nordvpn login command on your Linux device.
Open the provided link in any browser.
Complete the login procedure.
Right-click on the Return to the app button and select "Copy link address".
Run nordvpn login --callback <URL> with the previously copied URL.
Verify that login was successful with nordvpn account

**Comment by RiDDiX** [comment]
[services.d] done.
Invalid Username or password.
[cmd] /bin/sh exited 1
[cont-finish.d] executing container finish scripts...
[cont-finish.d] done.
[s6-finish] waiting for services.
[s6-finish] sending all processes the TERM signal.
[s6-finish] sending all processes the KILL signal and exiting.

Same problem. It still persist. After now nearly one month without a problem it came back. Even with disabled MFA it happens.

Also how should I am able to "nordvpn -login" within the Docker when it instantly kills itself, because it says Invalid Username or password?

Please a fix would be so awesome.

**Comment by Rickexe** [comment]
I think @akagami-x is talking about the nordlynx container with the alternative login method. I actually have nordlynx running now. But I would rather have the nordvpn container.

**Comment by RiDDiX** [comment]
Even with the NordLynx Container I dont know how to obtain a Private-Key and so on... Maybe Ill switch to the other working NordVPN Docker... It is just with OpenVPN and WireGuard but anyway it works atm...

**Comment by Rickexe** [comment]
Optaining private key is pretty easy. Just follow this tutorial: https://forum.openwrt.org/t/instruction-config-nordvpn-wireguard-nordlynx-on-openwrt/89976

**Comment by AESIR93** [comment]
I haven't been able to obtain the Private Key either, I have everything setup on Docker so I don't think I can make use of the above guide.
I have tried using the command provided on the bubuntux/nordlynx dockerhub: `docker run --rm --cap-add=NET_ADMIN -e USER=XXX -e PASS=YYY bubuntux/nordvpn nord_private_key`
But I get this message: 
`Unable to access interface: Protocol not supported
Private Key:
############################################################
[cmd] nord_private_key exited 0`

Any Idea how to solve it?
Ps: I am a bit of a noob so sorry If what I ask is trivial.

**Comment by Rickexe** [comment]
Don't do it like that. I think I tried that at first but didn't work either don't remember. Follow the instructions from the link. I'm on phone I'll try writing tutorial xD.

Easiest way, at least how I did it, was booting ubuntu or any other desktop distro. I have second drive with ubuntu on it but you can do this on a live usb key as well. So once you have that booted up throw some apt update and upgrade at it. 

Then do this:

install following packages on your machine:

wireguard sudo apt install wireguard
curl sudo apt install curl
jq sudo apt install jq
nettools sudo apt install net-tools
nordvpn sh <(curl -sSf https://downloads.nordcdn.com/apps/linux/install.sh)

Then it tells you to run this command:

log in to your nordvpn app via this command:
sudo nordvpn login

You'll have to login with an external link but that actually didn't work for me. I tried using tab to find out what commands where available and found that after the command "sudo nordvpn login" if you try typing "us" and "pas" and the press tab you'll find the right commands to login like that. Pretty sure it was sudo nordvpn login username=longcodefromnordvpnaccount password=otherlongcodefromnordvpnaccount 

Then run this:

change technology to nordlynx
sudo nordvpn set technology nordlynx

connect to your preferred server
sudo nordvpn c nl #to connect Nederland as an example

and this command for private key
sudo wg show nordlynx private-key

hope this helps gotta get back to work break's over lol bye

**Comment by AESIR93** [comment]
Okey, will try it this way and will get back to you with whatever I get.
Thank you very much in advance for such detailed explanation!!

**Comment by WINOFFRG** [comment]
Hello! Facing the similar issue but I found the resolution probably. I `exec` into the VPN Container and did the following steps:
1. Disconnect from VPN
2. Download Update 
3. Upgrade Packages
4. Connect to VPN

This seemed to work for me clearly until yesterday, When after reconnecting to VPN. I get the `host not resolved` error.
I also tried to work on version 3.12.3 the latest one, But still the same error. 

![image](https://user-images.githubusercontent.com/61311751/151695551-d5dc7992-3db1-4542-a86f-860fc3cb31c5.png)
